### PR TITLE
Improve Crossgen2 map file emitter

### PR DESCRIPTION
**Pull request by trylek**
I have rewritten the map file emitter with two main goals in mind:

1) Fix the long-standing problem that map file wasn't showing true
RVA's of the individual symbols / nodes, making it harder to
correlate its information with in-memory R2R PE layout.

2) Add section size and statistics section roughly mimicking what
JanV presented in his recent analytic breakdown of R2R contents.

Thanks

Tomas

### CODE REVIEW

**Comment by trylek** [comment]
Example map files:

https://gist.github.com/trylek/9bea813601ab182e07b20db9c0165dd9
https://gist.github.com/trylek/4e8eb741d65a517b8d9d59b020430b8d

File: src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/ObjectWriter/MapFileBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+                    if (symbolIndex < _symbols.Count && MapFileItem.Comparer.Instance.Compare(node, _symbols[symbolIndex]) == 0)
+                    {
+                        MapFileSymbol symbol = _symbols[symbolIndex++];
+                        writer.Write($"{symbol.Name}");
+                    }
+                    writer.WriteLine($"  ({node.Name})");
+                }
+            }
+
+            foreach (MapFileNode node in _nodes)
```
Comment (trylek) [inline src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/ObjectWriter/MapFileBuilder.cs:262]: Hehe, this is a dummy leftover from v#1 node list. Will delete :-).

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Pull request merged**